core/search/min_match_disjunction.hpp

```diff
@@ -305,38 +289,40 @@ class min_match_disjunction
   }
 
   template<typename Iterator>
-  inline void push(Iterator begin, Iterator end) {
+  inline void push(Iterator begin, Iterator end) noexcept {
     // lambda here gives ~20% speedup on GCC
-    std::push_heap(begin, end, [this](const size_t lhs, const size_t rhs) noexcept {
-      assert(lhs < itrs_.size());
-      assert(rhs < itrs_.size());
-      const auto& lhs_it = itrs_[lhs];
-      const auto& rhs_it = itrs_[rhs];
-      const auto lhs_doc = lhs_it.value();
-      const auto rhs_doc = rhs_it.value();
-      return (lhs_doc > rhs_doc || (lhs_doc == rhs_doc && lhs_it.est > rhs_it.est));
-    });
+    std::push_heap(begin, end,
+                   [this](const size_t lhs, const size_t rhs) noexcept {
+                     assert(lhs < itrs_.size());
+                     assert(rhs < itrs_.size());
+                     const auto& lhs_it = itrs_[lhs];
+                     const auto& rhs_it = itrs_[rhs];
+                     const auto lhs_doc = lhs_it.value();
+                     const auto rhs_doc = rhs_it.value();
+                     return (lhs_doc > rhs_doc ||
+                             (lhs_doc == rhs_doc && lhs_it.est > rhs_it.est));
+                   });
   }
 
   template<typename Iterator>
-  inline void pop(Iterator begin, Iterator end) {
+  inline void pop(Iterator begin, Iterator end) noexcept {
     // lambda here gives ~20% speedup on GCC
```

Comment:
is lambda better than lambda with call member function? maybe always inline
I just think duplicate comparator code isn't good (easy to break)